src/presenter/trip-presenter.js

```diff
@@ -1,100 +1,161 @@
     });
 
-    this.#filtersComponent = new FiltersView(eventsModel.events);
-    this.#sortingComponent = new SortingView(this.#onSortingChange);
+    this.#addEventButtonComponent = new ButtonView({ buttonElement: addEventButtonElement, onClick: this.#onAddEventClick });
 
-    this.#AddEventButtonElement = headerContainerElement.querySelector('.trip-main__event-add-btn');
-    this.#AddEventButtonElement.addEventListener('click', this.#onEventAddButtonElementClick);
+    eventsModel.addObserver(this.#onModelsChange);
+    filterModel.addObserver(this.#onModelsChange);
   }
 
   init() {
-    this.#events = this.#eventsModel.events;
+    this.#filterPresenter.init();//! скорее всего при плучении данных с сервера будет первый init
     this.#render();
   }
 
-  #render() {
-    this.#renderInfo();
-    this.#renderFilter();
-    this.#renderEvents();
+  #clear({ isResetSortingType } = { isResetSortingType: false }) {
+    remove(this.#sortingComponent);
+
+    if (isResetSortingType) {
+      this.#currentSortingType = DEFAULT_SORTING_TYPE;
+    }
+
+    if (this.#emptyEventsMessageComponent) {
+      remove(this.#emptyEventsMessageComponent);
+    }
   }
 
-  #renderInfo() {
-    //! нужно будет вызывать при изменении данных
+  #render() {
     this.#infoPresenter.init();
+    this.#renderEvents();
   }
 
-  #renderFilter() {
-    //! после добавления нового события пересчитать фильтры и отрисовать заново, если не все фильтры были активны
-    render(this.#filtersComponent, this.#headerTripFiltersElement);
+  #renderSorting() {
+    this.#sortingComponent = new SortingView({ currentSortingType: this.#currentSortingType, onSortingChange: this.#onSortingChange });
+    render(this.#sortingComponent, this.#tripEventsElement);
   }
 
-  #renderEvents() {
-    //! нужно будет вызывать и при изменении фильтра, добавлении нового события
-    //! и сортировать и фильтровать дабавленное новое событие
-    if (this.#events.length) {
-      render(this.#sortingComponent, this.#tripEventsElement);
+  #renderEvents({ isRenderSorting, isApplyFilter } = { isRenderSorting: true, isApplyFilter: true }) {
+    const now = Date.now();
 
-      this.#events.filter(() => true); //! временно, как будет готова фильтрация, то отфильтровать this.#events.filter(filterEvents(this.#currentFilterType))
-      this.#events.sort(sortEvents[this.#currentSortingType]);
+    //! нужно будет проверить вызов при добавлении нового события, или не так!
+    //! и сортировать и фильтровать дабавленное новое событие, или не так!
+    //! при сортировке заново фильтровались
+    const filteredEvents = (!isApplyFilter) ? this.#events : filterEvents(this.#eventsModel.events, this.#filterModel.filterType, now);
+
+    this.#eventsPresenter.clear();
+
+    if (!filteredEvents.length) {
+      this.#renderEmptyEventsMessage();
+      return;
     }
 
-    this.#eventsPresenter.init(this.#events);
+    if (isRenderSorting) {
+      this.#renderSorting();
+    }
```

Comment:
renderDashboard() {
  
  if (isRenderSorting) {
     this.#renderSorting();
  }

  #this.renderEvents();
}